metricflow/test/inference/context/test_snowflake.py

```diff
@@ -0,0 +1,122 @@
+import json
+from unittest.mock import MagicMock, call
+
+import pandas as pd
+
+from metricflow.dataflow.sql_table import SqlTable
+from metricflow.dataflow.sql_column import SqlColumn, SqlColumnType
+from metricflow.inference.context.data_warehouse import TableProperties, ColumnProperties, DataWarehouseInferenceContext
+from metricflow.inference.context.snowflake import SnowflakeInferenceContextProvider
+
+
+def test_column_type_conversion() -> None:  # noqa: D
+    ctx_provider = SnowflakeInferenceContextProvider(client=MagicMock(), tables=[])
+
+    # known snowflake types
+    assert ctx_provider._column_type_from_show_columns_data_type("FIXED") == SqlColumnType.INTEGER
+    assert ctx_provider._column_type_from_show_columns_data_type("TEXT") == SqlColumnType.STRING
+    assert ctx_provider._column_type_from_show_columns_data_type("REAL") == SqlColumnType.FLOAT
+    assert ctx_provider._column_type_from_show_columns_data_type("BOOLEAN") == SqlColumnType.BOOLEAN
+    assert ctx_provider._column_type_from_show_columns_data_type("DATE") == SqlColumnType.DATETIME
+    assert ctx_provider._column_type_from_show_columns_data_type("TIMESTAMP_TZ") == SqlColumnType.DATETIME
+    assert ctx_provider._column_type_from_show_columns_data_type("TIMESTAMP_LTZ") == SqlColumnType.DATETIME
+    assert ctx_provider._column_type_from_show_columns_data_type("TIMESTAMP_NTZ") == SqlColumnType.DATETIME
+
+    # unknowns
+    assert ctx_provider._column_type_from_show_columns_data_type("BINARY") == SqlColumnType.UNKNOWN
+    assert ctx_provider._column_type_from_show_columns_data_type("TIME") == SqlColumnType.UNKNOWN
+
+
+def test_context_provider() -> None:  # noqa: D
+    # See for SHOW COLUMNS result dataframe spec:
+    # https://docs.snowflake.com/en/sql-reference/sql/show-columns.html
+    show_columns_result = pd.DataFrame(
+        {
+            "column_name": ["INTCOL", "STRCOL"],
+            "schema_name": ["SCHEMA", "SCHEMA"],
+            "table_name": ["TABLE", "TABLE"],
+            "database_name": ["DB", "DB"],
+            "data_type": [
+                json.dumps({"type": "FIXED", "nullable": False}),
+                json.dumps({"type": "TEXT", "nullable": True}),
+            ],
+        }
+    )
+
+    stats_result = pd.DataFrame(
+        {
+            "intcol_countdistinct": [10],
+            "intcol_min": [0],
+            "intcol_max": [10],
+            "intcol_countnull": [0],
+            "strcol_countdistinct": [40],
+            "strcol_min": ["aaaa"],
+            "strcol_max": ["zzzz"],
+            "strcol_countnull": [10],
+            "rowcount": [50],
+        }
+    )
+
+    client = MagicMock()
+    client.query = MagicMock()
+    client.query.side_effect = [show_columns_result, stats_result]
```

Comment:
So I'm not a huge fan of mocks in this case. I'd prefer a proper fake, where there was like a local isolated database we could query, but I don't think Python provides that, so the mocking you're doing here might be as close as it gets.

At any rate, this test really only checks the mechanics against the boundary of the Snowflake query API itself, which can be fine, but here I don't think it gets us very much. There isn't very much data manipulation on the results to convert them into the TableProperties and ColumnProperties constructs, and the internal logic of those classes is already tested elsewhere.

It's fine to leave this here, but I think you'll want an integration test at some point. However, that should probably be in the base warehouse class, and it should essentially be using the correct context provider depending on which engine is under test.